fusedrug_examples/interaction/drug_target/affinity_prediction/t5_mol/tokenizer/multi_tokenizer.py

```diff
@@ -0,0 +1,1304 @@
+
+        Args:
+            token_str (str): _description_
+
+        Raises:
+            Exception: if could not find the token in any of the internal tokenizers
+
+        Returns:
+            int: ID of the input token if it exists
+        """
+        return self.mtokenizer.get_token_id(token_str)
+        # return None
+
+    def __call__(
+        self,
+        sample_dict: NDict,
+        key_in: str,
+        key_out_tokenized_object: str = None,
+        key_out_tokens_ids: str = None,
+        key_out_attention_mask: str = None,
+        convert_attention_mask_to_bool: bool = True,
+    ) -> NDict:
+        """_summary_
+
+        Args:
+            sample_dict (NDict): _description_
+            key_in (str): a key to a list of tuples (type, input_str) that is the ModularTokenizer.encode input
+            key_out_tokenized_object (str, optional): _description_. Defaults to None.
+            key_out_tokens_ids (str, optional): _description_. Defaults to None.
+            key_out_attention_mask (str, optional): _description_. Defaults to None.
+            convert_attention_mask_to_bool (bool, optional): _description_. Defaults to True.
+
+        Raises:
+            Exception: _description_
+            Exception: _description_
+
+        Returns:
+            NDict: _description_
+        """
+        typed_input_list = sample_dict[key_in]
+        if not isinstance(typed_input_list, list):
+            raise Exception(
+                f"Expected key_in={key_in} to point to a list, and instead got a {type(typed_input_list)}. value={typed_input_list}"
+            )
+        encoded = self.mtokenizer.encode_list(typed_input_list)
+
+        if len(encoded.overflowing) > 0:
+            print(
+                f"Warning: FastTokenizer had to truncate sequence. Truncated {encoded.overflowing[0].tokens} for sample_id {get_sample_id(sample_dict)}"
+            )
+
+        if key_out_tokenized_object is not None:
+            # if requested, store the entire tokenizer.Encoding object (which provides access to attributes such as  [ids, type_ids, tokens, offsets, attention_mask, special_tokens_mask, overflowing])
+            sample_dict[key_out_tokenized_object] = encoded
+
+        if key_out_tokens_ids is not None:
+            sample_dict[key_out_tokens_ids] = encoded.ids
+
+        if key_out_attention_mask is not None:
+            sample_dict[key_out_attention_mask] = encoded.attention_mask
+            if convert_attention_mask_to_bool:
+                sample_dict[key_out_attention_mask] = [bool(x) for x in sample_dict[key_out_attention_mask]]
+
+        if (key_out_tokens_ids is None) and (key_out_tokenized_object is None):
+            warn(
+                "FastTokenizer Op got key_out_tokens_ids=None and key_out_tokenized_object=None, which means it will not modify anything in the sample. Is this intended?"
+            )
+
+        return sample_dict
+
+
+class ModularStringTokenizerOp(OpBase):
```

Comment:
Can't it be a flag in ModularMultiTokenizerOp?

Reply:
The Ops are here more as a usage examples, and this one is not implemented. It can definitely be a flag, or even just an argument, based on the type of which we call encode_list or just encode

Reply:
Updated the code to do just that. the encode part that accepts a string is not implemented yet, though, so passing a string will raise an exception.

Reply:
Cool,
Lets also open trello card for the string implementation and for any other option you want to implement in the future,